src/Honeybee_IntersectMasses.py

```diff
@@ -50,173 +50,68 @@
-                        newBuilding = rc.Geometry.Brep.JoinBreps(newBldgBreps, sc.doc.ModelAbsoluteTolerance)[0]
-                        building = newBuilding
-                    except:
-                        pass
-    
-    return building
+    changed = False
+
+    joinedLines = []
+    i = 0
+    for i in range(500):
+        tempBldg = building.Duplicate()
+        for face1 in building.Faces:
+            if face1.IsSurface:
+                # if it is a untrimmed surface just find intersection lines
+                intersectLines = rc.Geometry.Intersect.Intersection.BrepSurface(otherBldg, face1.DuplicateSurface(), tol)[1]
+            else:
+                # if it is a trimmed surface
+                edgesIdx = face1.AdjacentEdges()
+                edges = []
+                for ix in edgesIdx:
+                    edges.append(building.Edges.Item[ix])
+                crv = rc.Geometry.Curve.JoinCurves(edges, tol)
+                # potential bug - a surface has made of multiple crvs, but not likely
+                intersectLines = rc.Geometry.Intersect.Intersection.BrepBrep(rc.Geometry.Brep.CreatePlanarBreps(crv[0])[0], otherBldg, tol)[1]
+            temp = rc.Geometry.Curve.JoinCurves(intersectLines, tol)
+            joinedLines = [crv for crv in temp if rc.Geometry.Brep.CreatePlanarBreps(crv)]
+            if len(joinedLines) > 0:
+                newBrep = face1.Split(joinedLines, tol) # return None on Failure
+                if not newBrep: continue
+                if newBrep.Faces.Count > building.Faces.Count: 
+                    changed = True
+                    building = newBrep
+                    break
+        if tempBldg.Faces.Count == building.Faces.Count:
+            break
+    return building, changed
+
 
 
 def main(bldgMassesBefore):
-    
     buildingDict = {}
-    intersectedBldgs = []
+
     for bldgCount, bldg in enumerate(bldgMassesBefore):
         buildingDict[bldgCount] = bldg
-    
-    
-    for bldgNum, building in buildingDict.items():
-        try:
-            #Intersect the geomtry with all buildings in the list after it.
-            for otherBldg in  bldgMassesBefore[:bldgNum]:
-                building = intersectMasses(bldgNum, building, otherBldg)
-                #Update the dictionary with the new split geometry.
-                buildingDict[bldgNum] = building
-                bldgMassesBefore[bldgNum] = building
-            
-            #Intersect the geomtry with all buildings in the list before it.
-            for otherBldg in  bldgMassesBefore[bldgNum+1:]:
-                building = intersectMasses(bldgNum, building, otherBldg)
-                #Update the dictionary with the new split geometry.
-                buildingDict[bldgNum] = building
-                bldgMassesBefore[bldgNum] = building
-        except:
+    need_change = deque(buildingDict.keys())
+    
+    i = 0 # to prevent dead loop
+    while(len(need_change) > 0 and i < 10e2*len(bldgMassesBefore)):
```

Comment:
Again, I am not so sure what makes 10e2*len(bldgMassesBefore) a special number here.

Reply:
Same as before. It's just a large number that `i` will not reach if the loop works correctly. I can get rid of it if you think it is not very clear.